src/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs

```diff
@@ -35,8 +35,50 @@ public static class TaskTimeoutExtensions
             }
             else
             {
-                throw new TimeoutException();
+                throw new TimeoutException($"Task timed out after {millisecondsTimeout}");
             }
         }
+
+        public static async Task WhenAllOrAnyFailed(this Task[] tasks, int millisecondsTimeout)
+        {
+            var cts = new CancellationTokenSource();
+            Task task = tasks.WhenAllOrAnyFailed();
+            if (task == await Task.WhenAny(task, Task.Delay(millisecondsTimeout, cts.Token)).ConfigureAwait(false))
+            {
+                cts.Cancel();
+                await task.ConfigureAwait(false);
```

Comment:
I just realized I was being silly when I suggested this needed `ConfigureAwait`... the task is guaranteed to already be completed here, so the ConfigureAwait will be a nop.  But that's fine, we can leave it.  Sorry about that.